### ISSUE TITLE
Find out if any action items are needed for GA4 before July 1, 2023

### ISSUE THREAD

**Issue by Maroyafaied** [DOI-ONRR/nrrd#2648]
Dear DAP User, 

We hope this email finds you well. The DAP team has been very busy and we have important updates to share with you.

Google Analytics Universal Analytics (UA) is going away and being replaced by a new Google Analytics 4 platform (GA4). On July 1, 2023, all free (non-DAP) UA properties will stop processing data, so we strongly encourage you to migrate your free Universal Analytics accounts to GA4 as soon as possible. Key dates and details follow.

Immediate action: Migrate your non-DAP UA free accounts to GA4 before July 1, 2023

DAP UA data collection will continue for another year (until July 1, 2024)

DAP GA4 tracking begins August 1, 2023

Upcoming action: If your agency hosts the DAP code file locally on your own servers, you’ll need to update the file when we send out a new version in August. Most agencies already call the DAP code from  https://dap.digitalgov.gov/Universal-Federated-Analytics-Min.js, so, if this is the case at your agency then no action is needed.

First, please migrate your non-DAP UA free accounts before July 1. DAP does not manage or assist agencies with their free GA account migration (DAP manages a separate paid/premium version of Google Analytics and this deadline does not apply to paid accounts). This action must be completed before July 1, 2023 so you don’t lose your independent free GA website tracking from Universal Analytics. The good news is that you will retain Viewer access to your free UA account historical data and reports in the Universal Analytics interface for six months (until January 1, 2024). So if you wish to keep the historical UA data, you will need to export it before January 1, 2024. 

Second, the DAP UA account will continue to collect UA data and provide reporting until July 1, 2024 (Google extended its deadline for paid accounts to give us an extra 12 months). So as long as you have the DAP tracking code on your websites, your DAP UA data collection and reporting will continue without interruption for another year. 

Third, we are excited to announce that the DAP team plans to launch new GA4 tracking on August 1, 2023! GA4 will provide new insights into an integrated user journey with new metrics and data on content usage and engagement across websites AND mobile apps. Because the GA4 platform is so different from the Universal Analytics, the DAP team spent months developing a dual solution that provides tracking for both UA and GA4 as part of the same tag. The dual solution will help ensure that the DAP team provides continuity of historical data from the DAP UA account through July 1, 2024 while we continue to learn the world of GA4. 

Finally, we know that most agencies already call the DAP code from the central location we host at https://dap.digitalgov.gov/Universal-Federated-Analytics-Min.js, so no action is needed on their part to switch to DAP GA4. When we launch the new code, we’ll publish it to that location and it will automatically collect and send data to both the existing Universal Analytics property AND our new GA4 Property. However, if your website hosts the DAP code locally on your own servers, then you'll need to replace your existing DAP version with the new production dual tracking code. On launch day, we’ll send an email with a link to that code version for you to replace your current code.

We know that this is a lot of information and you will have many questions (just like we do). To help with training and learning, we’ll be setting up learning sessions and DAP office hours in the months to come to support you with these changes. Keep an eye on the DAP listserv for upcoming updates.  

In the meantime, please see the summary of important dates and some initial FAQs below. Reach out to dap@gsa.gov for further questions and assistance. 

Thanks for all your patience and partnership,

Marina Fox - Digital Analytics Program Manager

Jack Avery - Digital Analytics Analyst (Helpdesk)

IMPORTANT DATES AND MILESTONES

Action

Date

Migrate your UA free (non-DAP) accounts to GA4 (Agencies must perform migration of their free UA accounts to GA4, independent of DAP)

Before July 1, 2023.

Note: After July 1, 2023, Google will stop collecting UA free accounts data). Historical data and reports will remain available for viewing and exporting for 6 months until January 1, 2024 

Begin using your free GA4 accounts and using new data (post migration). 

July 1, 2023 (or whenever agencies migrate their UA account to GA4)

Note:  Since GA4 collection and tracking model collection is vastly different from UA, GA4 data collection will begin from scratch once migration is complete. No historical data will be migrated to GA4 from UA. 

ONLY If your agency plans to host the new DAP code vs. calling it from a central hosting location like most agencies already do: replace your existing DAP code version with the new dual tracking code 

August 1, 2023 

Export historical UA data and reports from free (non-DAP) accounts 

Before January 1, 2024

You won’t be able to access your previously processed historical data and view your saved reports after January 1, 2024. 

Q: Will agencies lose access to DAP Universal Analytics (UA) as soon as GA4 is launched?

A: No. The DAP Universal Analytics (UA) account will continue to run in parallel with the new GA4 account until July 1, 2024. You will have access to both DAP Universal Analytics and DAP GA4 accounts until July 1, 2024 to help with the learning curve associated with GA4. 

Q: Will the DAP GA4 launch affect the DAP UA data currently used?

A: No. The new code was developed to ensure that the current DAP UA data collection is not affected. If you notice any anomalies, please email dap@gsa.gov.

**Comment by Maroyafaied** [comment]
@cthomasONRR Sure, it doesn't hurt to confirm!

**Comment by cthomasONRR** [comment]
I emailed Jack. He confirmed that DAP code is present and no action is required.

He did point out that for revenuedata.doi.gov and blog-nrrd.doi.gov, 
the DAP implementation on these pages appears to be non-standard. In particular, there is no 'agency' parameter being passed by these pages. Please review the implementation guidelines for including the 'agency' parameter located in the Code Capabilities Summary at the [DAP GitHub Repo](https://github.com/digital-analytics-program/gov-wide-code). This is not related to the GA4 transition.

I can make a new issue for this.

**Comment by cthomasONRR** [comment]
I created the following issue to address the non-standard DAP code in NRRD and the blog: https://github.com/DOI-ONRR/nrrd/issues/2661

**Comment by Maroyafaied** [comment]
@cthomasONRR Thanks for contacting them. I'm not sure what the agency parameter is but  we can have Jeff take a look next sprint.

**Comment by cthomasONRR** [comment]
@Maroyafaied OK I added the issue to next sprint and added Jeff to it.

**Comment by Maroyafaied** [comment]
Closing this since we confirmed there's no action needed.
